From 9b4d36409fbde4c165051738e1f1ed908d5ab40c Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Wed, 1 Nov 2017 22:14:11 +0800
Subject: [PATCH 02/10] video: sunxi: de2: add support for LCD SimpleFB

Add support for setting up SimpleFB for LCD display output in DE2
SimpleFB setup code.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 drivers/video/sunxi/sunxi_de2.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/drivers/video/sunxi/sunxi_de2.c b/drivers/video/sunxi/sunxi_de2.c
index 6d6bb2e0c3..860187323c 100644
--- a/drivers/video/sunxi/sunxi_de2.c
+++ b/drivers/video/sunxi/sunxi_de2.c
@@ -319,7 +319,7 @@ U_BOOT_DEVICE(sunxi_de2) = {
 #if defined(CONFIG_OF_BOARD_SETUP) && defined(CONFIG_VIDEO_DT_SIMPLEFB)
 int sunxi_simplefb_setup(void *blob)
 {
-	struct udevice *de2, *hdmi;
+	struct udevice *de2, *hdmi, *lcd;
 	struct video_priv *de2_priv;
 	struct video_uc_platdata *de2_plat;
 	int mux;
@@ -355,6 +355,15 @@ int sunxi_simplefb_setup(void *blob)
 		debug("HDMI present but not probed\n");
 	}
 
+	ret = uclass_find_device_by_name(UCLASS_DISPLAY,
+					 "sunxi_lcd", &lcd);
+	if (ret)
+		debug("LCD not present\n");
+	else if (device_active(lcd))
+		pipeline = "mixer0-lcd0";
+	else
+		debug("LCD present but not probed\n");
+
 	if (!pipeline) {
 		debug("No active display present\n");
 		return 0;
-- 
2.13.6

